addons/account/report/account_aged_partner_balance.py

```diff
@@ -49,25 +49,23 @@ def _get_partner_move_lines(self, account_type, date_from, target_move, period_l
         move_state = ['draft', 'posted']
         if target_move == 'posted':
             move_state = ['posted']
-        arg_list = (tuple(move_state), tuple(account_type))
-        #build the reconciliation clause to see what partner needs to be printed
-        reconciliation_clause = '(l.reconciled IS FALSE)'
-        cr.execute('SELECT debit_move_id, credit_move_id FROM account_partial_reconcile where max_date > %s', (date_from,))
-        reconciled_after_date = []
-        for row in cr.fetchall():
-            reconciled_after_date += [row[0], row[1]]
-        if reconciled_after_date:
-            reconciliation_clause = '(l.reconciled IS FALSE OR l.id IN %s)'
-            arg_list += (tuple(reconciled_after_date),)
-        arg_list += (date_from, tuple(company_ids))
+        arg_list = (tuple(move_state), tuple(account_type),
+                    date_from, date_from, date_from, tuple(company_ids))
         query = '''
             SELECT DISTINCT l.partner_id, UPPER(res_partner.name)
             FROM account_move_line AS l left join res_partner on l.partner_id = res_partner.id, account_account, account_move am
             WHERE (l.account_id = account_account.id)
                 AND (l.move_id = am.id)
                 AND (am.state IN %s)
                 AND (account_account.internal_type IN %s)
-                AND ''' + reconciliation_clause + '''
+                AND (
+                        l.reconciled IS FALSE
+                        OR l.id IN(
+                            SELECT credit_move_id FROM account_partial_reconcile where max_date > %s
+                            UNION ALL
+                            SELECT debit_move_id FROM account_partial_reconcile where max_date > %s
```

Comment:
Instead of using `UNION ALL` on 2 queries, can you test (correctness and execution time) with
```sql
SELECT unnest(ARRAY[credit_move_id, debit_move_id]) FROM account_partial_reconcile where max_date > %s
```
(Note: do not forget to remove one `date_from` from `arg_list`)

This will actually only make 1 seq scan instead of 2.

Reply:
Thank you for reviewing this! 

After changing arg_list and replacing the subquery with your suggestion, the "shorter" execution time (report for today) takes > 60s, and the "longer" one (report for Oct 31st) takes > 800s. Is there an explanation, or did I forget to change something? I retested with nse's orginal changes, and it is always around 30s.

Reply:
With the first change, the query is quite fast:
![unionall](https://user-images.githubusercontent.com/46444466/70309643-408fdc00-180e-11ea-881e-20209b0b954f.png)

After the second change, the query takes > 10 minutes (the explain analyze would also take a while)
![unnest](https://user-images.githubusercontent.com/46444466/70309651-44236300-180e-11ea-9943-3c9ec1e5bd8b.png)

Reply:
I still don't understand why it still so slow. So, let's keep the actual patch.

@robodoo r+

Reply:
Thanks, if you would like more tests to understand what happens with unnest let me know.